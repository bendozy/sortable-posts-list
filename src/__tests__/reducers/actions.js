import reducer, { initialState } from '../../reducers/actions';
import {
  REMOVE_LAST_ACTION,
  MOVE_POST,
  DOWN,
  UP,
} from '../../constants';

describe('Actions reducer', () => {
  it('should return the initial state', () => {
    const result = reducer(undefined, {});

    expect(result).toEqual(initialState);
    expect(result).toMatchSnapshot();
  });

  it('should remove the last action', () => {
    const action = {
      type: REMOVE_LAST_ACTION,
    };
    const currentState = {
      actions: [{
        title: 'hello',
        from: 2,
        to: 1,
        key: 15,
      }],
      nextKey: 1,
    };

    const result = reducer(currentState, action);

    expect(result).toEqual({
      actions: [],
      nextKey: 0,
    });
    expect(result).toMatchSnapshot();
  });

  it('should add a moved down post action', () => {
    const action = {
      type: MOVE_POST,
      payload: {
        direction: DOWN,
        index: 0,
        post: {
          title: 'hello',
        },
      },
    };
    const expectedAction = {
      title: 'hello',
      from: 0,
      to: 1,
      key: 0,
    };

    const result = reducer(initialState, action);

    expect(result).toEqual({
      actions: [expectedAction],
      nextKey: 1,
    });
    expect(result).toMatchSnapshot();
  });

  it('should add a moved up post action', () => {
    const action = {
      type: MOVE_POST,
      payload: {
        direction: UP,
        index: 1,
        post: {
          title: 'hello',
        },
      },
    };
    const expectedAction = {
      title: 'hello',
      from: 1,
      to: 0,
      key: 0,
    };

    const result = reducer(initialState, action);

    expect(result).toEqual({
      actions: [expectedAction],
      nextKey: 1,
    });
    expect(result).toMatchSnapshot();
  });
});
