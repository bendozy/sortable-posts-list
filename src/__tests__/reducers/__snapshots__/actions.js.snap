// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions reducer should add a moved down post action 1`] = `
Object {
  "actions": Array [
    Object {
      "from": 0,
      "key": 0,
      "title": "hello",
      "to": 1,
    },
  ],
  "nextKey": 1,
}
`;

exports[`Actions reducer should add a moved up post action 1`] = `
Object {
  "actions": Array [
    Object {
      "from": 1,
      "key": 0,
      "title": "hello",
      "to": 0,
    },
  ],
  "nextKey": 1,
}
`;

exports[`Actions reducer should remove the last action 1`] = `
Object {
  "actions": Array [],
  "nextKey": 0,
}
`;

exports[`Actions reducer should return the initial state 1`] = `
Object {
  "actions": Array [],
  "nextKey": 0,
}
`;
